src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -303,10 +303,12 @@ private OperationStatus CheckActiveStatements(IEnumerable<StatementSyntax> state
             private IEnumerable<StatementSyntax> MoveDeclarationOutFromMethodDefinition(
                 IEnumerable<StatementSyntax> statements, CancellationToken cancellationToken)
             {
-                var variableToRemoveMap = CreateVariableDeclarationToRemoveMap(
-                    this.AnalyzerResult.GetVariablesToMoveOutToCallSiteOrDelete(cancellationToken), cancellationToken);
+                var variablesToRemove = this.AnalyzerResult.GetVariablesToMoveOutToCallSiteOrDelete(cancellationToken);
+                var variableToRemoveMap = CreateVariableDeclarationToRemoveMap(variablesToRemove, cancellationToken);
+                var variableNamesToRemove = new HashSet<string>(variablesToRemove.Select(v => v.Name));
 
-                foreach (var statement in statements)
+                foreach (var statement in statements
```

Comment:
in general, avoid complex expressions that make statement headers tough to read.  In this case, just introduce a location for "statements.Select(...)"